Code/Tools/ProjectManager/Source/PythonBindings.cpp

```diff
@@ -250,6 +236,29 @@ namespace O3DE::ProjectManager
         StopPython();
     }
 
+    void PythonBindings::OnStdOut(const char* msg)
+    {
+        AZStd::string message{ msg };
+        // escape % characters by using %% in the format string to avoid crashing
+        AZ::StringFunc::Replace(message, "%", "%%", true /* case sensitive since it is faster */);
+        AZ_TracePrintf("Python", message.c_str());
+    }
+
+    void PythonBindings::OnStdError(const char* msg)
+    {
+        AZStd::string lastPythonError = msg;
+        constexpr const char* pythonErrorPrefix = "ERROR:root:";
+        constexpr size_t lengthOfErrorPrefix = AZStd::char_traits<char>::length(pythonErrorPrefix);
+        auto errorPrefix = lastPythonError.find(pythonErrorPrefix);
+        if (errorPrefix != AZStd::string::npos)
+        {
+            lastPythonError.erase(errorPrefix, lengthOfErrorPrefix);
+        }
+        O3DE::ProjectManager::PythonBindingsInterface::Get()->AddErrorString(lastPythonError);
```

Comment:
If you change lastPythonError to a string_view, then it wouldn't need to perform any memory allocations.
Also the substring would be faster.

Reply:
Should be addressed in #10168